### ISSUE TITLE
0.4.22 fails to build as a dependency in docs.rs

### ISSUE THREAD

**Issue by dtolnay** [alexcrichton/proc-macro2#147]
> I'm having trouble building the docs for my crate due to this issue it seems: https://docs.rs/crate/cqc/0.3.4/builds/126609
>
> I can't reproduce the error on my machine on neither stable or nightly. Any ideas what's going wrong?

_Originally posted by @Wojtek242 in https://github.com/alexcrichton/proc-macro2/issues/145#issuecomment-437712967_

**Comment by dtolnay** [comment]
I don't understand what happened in this build. Rustc is being invoked with the flags:

```bash
--cfg feature="default" \
--cfg feature="proc-macro" \
--cfg u128 \
--cfg use_proc_macro \
--cfg wrap_proc_macro \
--cfg super_unstable
```

Super_unstable is only defined here, and only if procmacro2_semver_exempt is defined:

https://github.com/alexcrichton/proc-macro2/blob/c239f03ca8a4259cb8d164d7e8a5fe18e7e38be2/build.rs#L28-L30

But I don't see an invocation of our build script in the cqc build log. It's almost like the rustc invocation is cached from a previous build. In our case that's not what we want, because we want to enable unstable features when building https://docs.rs/proc-macro2 but not when proc-macro2 is used as a dependency. Otherwise we end up breaking downstream crates that don't require nightly (as seen here) whenever APIs change in nightly.

As a shot in the dark, I think d7568e53327c614200ccf5d4b61bbbc62e566780 may fix that failure. I published 0.4.23 and it should get picked up if you publish your crate again. Once you're unblocked I will file a follow-up issue to figure out how we can avoid bringing unstable features into docs builds of downstream crates, while still having our own docs in docs.rs display all the unstable APIs.

**Comment by Wojtek242** [comment]
It seems to work now :) https://docs.rs/cqc/0.3.5/cqc/

Thanks a lot!

**Comment by dtolnay** [comment]
Nice. Sorry about the breakage.

I filed #148 to follow up on this unstable features documentation situation.
